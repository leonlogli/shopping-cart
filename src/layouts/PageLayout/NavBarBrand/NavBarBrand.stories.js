import React from 'react'
import NavBarBrand from '.'

export default {
  title: 'NavBarBrand',
}

export const Default = () => <NavBarBrand />

export const withName = () => <NavBarBrand name="Home" />
