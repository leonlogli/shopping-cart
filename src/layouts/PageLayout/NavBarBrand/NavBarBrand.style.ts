import styled from 'styled-components'

import { Box } from '../../../components'

const NavBarBrandRoot = styled(Box)`
  a:first-child {
    line-height: 0;
  }

  a:nth-last-child(1) {
    color: inherit;
    margin-left: 10px;
    font-weight: 700;
    font-size: 20px;
    line-height: 20px;
  }
`

export { NavBarBrandRoot }
export default NavBarBrandRoot
