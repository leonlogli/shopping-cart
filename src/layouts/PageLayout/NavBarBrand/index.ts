export * from './NavBarBrand'
export { default } from './NavBarBrand'
