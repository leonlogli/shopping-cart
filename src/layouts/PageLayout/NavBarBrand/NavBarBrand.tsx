import React from 'react'

import { NavBarBrandRoot } from './NavBarBrand.style'
import { Link, BoxProps, Logo, LogoProps } from '../../../components'

export type NavBarBrandProps = BoxProps & {
  /**
   * Nav brand name
   * @default Recipoz
   */
  name?: string
  logoProps?: LogoProps
}

const NavBarBrand = (props: NavBarBrandProps) => {
  const { name = 'Shopping Cart', logoProps, ...other } = props

  return (
    <NavBarBrandRoot display="flex" alignItems="center" {...other}>
      <Link underline="none" href="/">
        <Logo
          size={logoProps?.size || '26'}
          fill={logoProps?.fill}
          color={logoProps?.color}
        />
      </Link>
      <Link underline="none" href="/">
        {name}
      </Link>
    </NavBarBrandRoot>
  )
}

export { NavBarBrand }
export default NavBarBrand
