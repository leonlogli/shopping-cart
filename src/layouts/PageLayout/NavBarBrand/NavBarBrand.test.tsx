import React from 'react'
import { mount } from 'enzyme'

import NavBarBrand from '.'

describe('<NavBarBrand />', () => {
  const wrapper = mount(<NavBarBrand />)

  it('should render children', () => {
    expect(wrapper.find('a').length).toEqual(2)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
