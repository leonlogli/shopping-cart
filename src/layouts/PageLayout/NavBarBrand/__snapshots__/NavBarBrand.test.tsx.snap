// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarBrand /> should match a snapshot 1`] = `
<NavBarBrand>
  <Styled(Component)
    alignItems="center"
    display="flex"
  >
    <ForwardRef
      alignItems="center"
      className="ccsczA"
      display="flex"
    >
      <Styled(Styled(MuiBox))
        alignItems="center"
        className="ccsczA"
        display="flex"
      >
        <Styled(MuiBox)
          alignItems="center"
          className="dUNeVP ccsczA"
          display="flex"
        >
          <div
            className="MuiBox-root MuiBox-root-1 dUNeVP ccsczA"
          >
            <ForwardRef
              href="/"
              underline="none"
            >
              <Link
                href="/"
                passHref={true}
              >
                <Styled(WithStyles(ForwardRef(Link)))
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  underline="none"
                >
                  <WithStyles(ForwardRef(Link))
                    className="jxeIVE"
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    underline="none"
                  >
                    <ForwardRef(Link)
                      className="jxeIVE"
                      classes={
                        Object {
                          "button": "MuiLink-button",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiLink-root",
                          "underlineAlways": "MuiLink-underlineAlways",
                          "underlineHover": "MuiLink-underlineHover",
                          "underlineNone": "MuiLink-underlineNone",
                        }
                      }
                      href="/"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      underline="none"
                    >
                      <WithStyles(ForwardRef(Typography))
                        className="MuiLink-root MuiLink-underlineNone jxeIVE"
                        color="primary"
                        component="a"
                        href="/"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        variant="inherit"
                      >
                        <ForwardRef(Typography)
                          className="MuiLink-root MuiLink-underlineNone jxeIVE"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="primary"
                          component="a"
                          href="/"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          variant="inherit"
                        >
                          <a
                            className="MuiTypography-root MuiLink-root MuiLink-underlineNone jxeIVE MuiTypography-colorPrimary"
                            href="/"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <ForwardRef
                              size="26"
                            >
                              <svg
                                className="Logo "
                                fill="white"
                                height="26"
                                viewBox="0 0 24 24"
                                width="26"
                              >
                                <path
                                  d="M2,16H22V18C22,19.11 21.11,20 20,20H4C2.89,20 2,19.11 2,18V16M6,4H18C20.22,4 22,5.78 22,8V10H2V8C2,5.78 3.78,4 6,4M4,11H15L17,13L19,11H20C21.11,11 22,11.89 22,13C22,14.11 21.11,15 20,15H4C2.89,15 2,14.11 2,13C2,11.89 2.89,11 4,11Z"
                                />
                              </svg>
                            </ForwardRef>
                          </a>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </ForwardRef(Link)>
                  </WithStyles(ForwardRef(Link))>
                </Styled(WithStyles(ForwardRef(Link)))>
              </Link>
            </ForwardRef>
            <ForwardRef
              href="/"
              underline="none"
            >
              <Link
                href="/"
                passHref={true}
              >
                <Styled(WithStyles(ForwardRef(Link)))
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  underline="none"
                >
                  <WithStyles(ForwardRef(Link))
                    className="jxeIVE"
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    underline="none"
                  >
                    <ForwardRef(Link)
                      className="jxeIVE"
                      classes={
                        Object {
                          "button": "MuiLink-button",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiLink-root",
                          "underlineAlways": "MuiLink-underlineAlways",
                          "underlineHover": "MuiLink-underlineHover",
                          "underlineNone": "MuiLink-underlineNone",
                        }
                      }
                      href="/"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      underline="none"
                    >
                      <WithStyles(ForwardRef(Typography))
                        className="MuiLink-root MuiLink-underlineNone jxeIVE"
                        color="primary"
                        component="a"
                        href="/"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        variant="inherit"
                      >
                        <ForwardRef(Typography)
                          className="MuiLink-root MuiLink-underlineNone jxeIVE"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="primary"
                          component="a"
                          href="/"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          variant="inherit"
                        >
                          <a
                            className="MuiTypography-root MuiLink-root MuiLink-underlineNone jxeIVE MuiTypography-colorPrimary"
                            href="/"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                          >
                            Shopping Cart
                          </a>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </ForwardRef(Link)>
                  </WithStyles(ForwardRef(Link))>
                </Styled(WithStyles(ForwardRef(Link)))>
              </Link>
            </ForwardRef>
          </div>
        </Styled(MuiBox)>
      </Styled(Styled(MuiBox))>
    </ForwardRef>
  </Styled(Component)>
</NavBarBrand>
`;
